src/Credentials/InstanceProfileProvider.php

```diff
@@ -2,6 +2,7 @@
 namespace Aws\Credentials;
```

Comment:
I don't think this credential provider should have any knowledge of a cache at all. Caching should be a wrapper around other providers.